public/contexts/PermissionsContext.js

```diff
@@ -31,35 +31,48 @@ export function PermissionsProvider({ children }) {
   const [permissions, setPermissions] = useState([]);
   const { token, loggedIn } = useAuthToken();
 
-  const putPermissions = useCallback((newPermissions) => {
-    setPermissions(newPermissions);
-    localStorage.setItem("permissions", JSON.stringify(newPermissions));
-  }, []);
-
-  const clearPermissions = useCallback(() => {
+  const removePermissions = useCallback(() => {
     setPermissions(null);
     localStorage.removeItem("permissions");
   }, []);
 
+  const putPermissions = useCallback(
+    (newPermissions) => {
+      if (
+        !newPermissions ||
+        newPermissions === "undefined" ||
+        newPermissions === "null"
+      ) {
+        removePermissions();
+      } else {
+        setPermissions(newPermissions);
+        localStorage.setItem("permissions", JSON.stringify(newPermissions));
+      }
+    },
+    [removePermissions],
+  );
+
   const permissionsCheck = useCallback(
     (page, right) => {
       if (process.env.PERMISSIONS_DISABLED === "true") {
         return true;
       }
 
-      if (permissions && loggedIn) {
-        return findPermission(permissions, page, right);
-      }
-
-      return false;
+      return permissions && loggedIn
+        ? findPermission(permissions, page, right)
+        : false;
     },
     [permissions, loggedIn],
   );
 
   useEffect(() => {
     const setPermissionsOnLoad = () => {
       const permissionsStored = localStorage.getItem("permissions");
-      if (permissionsStored) {
+      if (
+        permissionsStored ||
+        permissionsStored !== "undefined" ||
```

Comment:
probably needs to be an && operation instead of an || operation?

Reply:
You're correct, thanks! I added a helper function `storeValueIsUndefined` that I use here and in other places